src/gluonts/dataset/parallelized_loader.py

```diff
@@ -346,6 +355,45 @@ def _worker_fn(
     return buf.getvalue()
 
 
+class _ShuffleBatchIter(Iterator[DataEntry]):
```

Comment:
You can call this `ShuffleIterator` (without underscore, and without “batch” since this object knows nothing about batches)